### DISCUSSION TITLE
Is there an API call for pending requests?

### DISCUSSION THREAD

**Discussion by gianpietro1** [AzuraCast/AzuraCast#6046] [Q&A]
I've noticed you can query the list of "requestable" songs, you can also query the queue, but it seems there is not a call for "pending requests" (requested but not yet queued). Is this true? or it is just missing the documentation at swagger?
Thanks.

**Comment by guiltlab** [comment]
There is! Actually I've been working on this myself very recently, so I have the solution, you're lucky :D 
You will need to use an API key. Here is what I use:

```
axios.get(${api}/station/${stationId}/reports/requests?type=pending, {
                headers: {
                    "X-Api-Key": apikey
                },
            })
```

There are quite a few API endpoints that are not documented on the Swagger. As per Buster & Vaalyn on Discord when I asked, pretty much everything you can do in the internal admin pages is linked to an API endpoint. You can use dev tools (F12) on your browser to find them, using the Network tab. For example, the way I found the correct URL for pending requests (as well as request history): 
- went to https://my.site/station/1/reports/requests
- clicked on "pending requests" and "request history" tabs, look what happens in the network dev tools
- a GET request to https://my.site/api/station/1/reports/requests?type=pending&internal=true&rowCount=0 and https://my.site/api/station/1/reports/requests?type=history appeared in the network tab
- answer found!

Cheers

**Reply by gianpietro1** [reply]
That was great! I was able to implement some visibility for the pending requests in my app.
Thanks so much, and for the extra info! :)
